api/include/opentelemetry/_metrics/async_instruments.h

```diff
@@ -4,8 +4,8 @@
 #pragma once
 #ifdef ENABLE_METRICS_PREVIEW
 
-#  include "instrument.h"
-#  include "observer_result.h"
+#  include "opentelemetry/_metrics/instrument.h"
```

Comment:
Could we add implication that the folder `_metrics` is deprecated, like name it as `metrics_deprecated`?

Reply:
That's a good point. I started with `metrics_deprecated` and it looked weirdly long to me including as `#include` in headers and srcs. I borrowed `_metrics` directory notation from python SIG the way they are handling it. 
To make it more clear, have updated the status in README.md along with specifying this directory name.